hardware/ariadne/bootloaders/ariadne/tftp.c

```diff
@@ -37,7 +37,13 @@ const unsigned char tftp_unknown_error_packet[] PROGMEM = "\0\5" "\0\0" "Error";
 #define TFTP_INVALID_IMAGE_LEN 23
 const unsigned char tftp_invalid_image_packet[] PROGMEM = "\0\5" "\0\0" "Invalid image file";
 
-uint16_t lastPacket = 0, highPacket = 0;
+uint8_t tftpFlashing;
+
+#ifndef TFTP_RANDOM_PORT
+static uint16_t tftpTransferPort;
+#endif
+
+static uint16_t lastPacket = 0, highPacket = 0;
```

Comment:
i think you confuse static variable in method/function with static keyword for variable on file scope (translation unit), which in this case is for his visibility (local vs global).

same word different meaning, this why c++ programmers don't like it, and prefer using anonymous namespace instead of static keyword for file's scope variables.

many, many discussion on this theme found in stackoverflow.com